### PR TITLE
Add pull_request_target event to PR workflows.

### PR DESCRIPTION
**Pull request by rin-skylight**
# DEVOPS PULL REQUEST

## Related Issue

- Our repository is paradoxically configured. As things stand, a series of completed checks are required for a PR to be merged. However, PRs that originate from an outside fork do not trigger the PR checks, as the source branches are not part of our existing code. Because of this, PRs from forks can never be merged, as the checks will never run.

## Changes Proposed

- Leverage the `pull_request_target` event as a valid trigger for PR check workflows.
 
### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)

### CODE REVIEW

**Comment by emmastephenson** [comment]
@rin-skylight can you clarify what you mean by an outside fork? Is that for PRs that are not merging to `main` but rather to a separate branch?

**Comment by rin-skylight** [comment]
@emmastephenson In this case, "outside fork" would mean a fork of the entire repository by an outside contributor to one of their own personal repositories. 

For anyone not directly affiliated with the project, we request that they create one of these forks and merge from it, rather than branching directly off of our project. More information is available in our maintainers documentation.

**Review by emmastephenson — APPROVED** [review]

**Comment by alismx** [comment]
As far as I understand how running workflows from a fork goes, it requires approval from a repo admin. Does this remove the approval process required to run those workflows or is it simply enabling them to be run at all?

**Review by alismx — APPROVED** [review]

**Comment by rin-skylight** [comment]
> As far as I understand how running workflows from a fork goes, it requires approval from a repo admin. Does this remove the approval process needed to run those workflows, or is it simply enabling them to be run against a fork at all?

Great question! To my knowledge, this will simply enable them to be run against the fork. It should not remove the approval process. That said...I will wait to merge this during business hours, in case I am wrong. That way, we can get a quick reversion.

**Comment by zdeveloper** [comment]
@rin-skylight any update on this PR? does it need to be merged or closed?

**Pull request closed without merging**
